{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This note book is for converting BioCXML format to IOB2\n",
    "##### salma, 9th of april\n",
    "##### Modular python code is written as BioCXML_to_IOB2_coversion_tool.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n",
      "<!DOCTYPE collection SYSTEM \"BioC.dtd\">\n",
      "\n",
      "<collection>\n",
      "<source/>Pubmed\n",
      "  <date>20200306</date>\n",
      "<key></key>\n",
      "<infon key=\"BQ_URL\">http://www.bioqrator.org</infon>\n",
      "<infon key=\"BQ_NAME\">Corona_gold</infon>\n",
      "<document>\n",
      "<id>31991541</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.4251</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>Return of the Coronavirus: 2019-nCoV. </text>\n",
      "<annotation id=\"A4\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"27\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A5\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"NCBI:txid\">693995</infon>\n",
      "<location length=\"11\" offset=\"14\"></location>\n",
      "<text>Coronavirus</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>38</offset>\n",
      "<text>The emergence of a novel coronavirus (2019-nCoV) has awakened the echoes of SARS-CoV from nearly two decades ago. Yet, with technological advances and important lessons gained from previous outbreaks, perhaps the world is better equipped to deal with the most recent emergent group 2B coronavirus. </text>\n",
      "<annotation id=\"A2\">\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<infon key=\"NCBI:txid\">694009 </infon>\n",
      "<location length=\"8\" offset=\"114\"></location>\n",
      "<text>SARS-CoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A3\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"41\" offset=\"293\"></location>\n",
      "<text>most recent emergent group 2B coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A1\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"17\" offset=\"57\"></location>\n",
      "<text>novel coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A6\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"76\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>32013309</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.6238</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>Mast cells contribute to coronavirus-induced inflammation: new anti-inflammatory strategy. </text>\n",
      "<annotation id=\"A65\">\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"NCBI:txid\">693995</infon>\n",
      "<location length=\"11\" offset=\"25\"></location>\n",
      "<text>coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A66\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"12\" offset=\"45\"></location>\n",
      "<text>inflammation</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>91</offset>\n",
      "<text>Coronavirus, which can cause respiratory syndrome, to date has affected over seventeen thousand individuals, especially in China. Coronavirus is interspecies and can also be transmitted from man to man, with an incubation ranging from 1 to 14 days. Human coronavirus infections can induce not only mild to severe respiratory diseases, but also inflammation, high fever, cough, acute respiratory tract infection and dysfunction of internal organs that may lead to death. Coronavirus infection (regardless of the various types of corona virus) is primarily attacked by immune cells including mast cells (MCs), which are located in the submucosa of the respiratory tract and in the nasal cavity and represent a barrier of protection against microorganisms. Virus activate MCs which release early inflammatory chemical compounds including histamine and protease; while late activation provokes the generation of pro-inflammatory IL-1 family members including IL-1 and IL-33. Here, we propose for the first time that inflammation by coronavirus may be inhibited by anti-inflammatory cytokines belonging to the IL-1 family members. </text>\n",
      "<annotation id=\"A67\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"11\" offset=\"91\"></location>\n",
      "<text>Coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A68\">\n",
      "<infon key=\"representative_name\">respiratory disease</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"20\" offset=\"120\"></location>\n",
      "<text>respiratory syndrome</text>\n",
      "</annotation>\n",
      "<annotation id=\"A69\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"NCBI:txid\">693995</infon>\n",
      "<location length=\"11\" offset=\"221\"></location>\n",
      "<text>Coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A70\">\n",
      "<infon key=\"representative_name\">coronavirus infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"28\" offset=\"340\"></location>\n",
      "<text>Human coronavirus infections</text>\n",
      "</annotation>\n",
      "<annotation id=\"A71\">\n",
      "<infon key=\"representative_name\">respiratory disease</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"20\" offset=\"404\"></location>\n",
      "<text>respiratory diseases</text>\n",
      "</annotation>\n",
      "<annotation id=\"A72\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"12\" offset=\"435\"></location>\n",
      "<text>inflammation</text>\n",
      "</annotation>\n",
      "<annotation id=\"A73\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"454\"></location>\n",
      "<text>fever</text>\n",
      "</annotation>\n",
      "<annotation id=\"A74\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"461\"></location>\n",
      "<text>cough</text>\n",
      "</annotation>\n",
      "<annotation id=\"A75\">\n",
      "<infon key=\"representative_name\">respiratory disease</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"27\" offset=\"474\"></location>\n",
      "<text>respiratory tract infection</text>\n",
      "</annotation>\n",
      "<annotation id=\"A77\">\n",
      "<infon key=\"representative_name\">organ dysfunction</infon>\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"30\" offset=\"506\"></location>\n",
      "<text>dysfunction of internal organs</text>\n",
      "</annotation>\n",
      "<annotation id=\"A78\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"554\"></location>\n",
      "<text>death</text>\n",
      "</annotation>\n",
      "<annotation id=\"A79\">\n",
      "<infon key=\"representative_name\">coronavirus infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"21\" offset=\"561\"></location>\n",
      "<text>Coronavirus infection</text>\n",
      "</annotation>\n",
      "<annotation id=\"A80\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"NCBI:txid\">693995</infon>\n",
      "<location length=\"12\" offset=\"619\"></location>\n",
      "<text>corona virus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A81\">\n",
      "<infon key=\"representative_name\">virus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<location length=\"5\" offset=\"845\"></location>\n",
      "<text>Virus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A83\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"12\" offset=\"1103\"></location>\n",
      "<text>inflammation</text>\n",
      "</annotation>\n",
      "<annotation id=\"A84\">\n",
      "<infon key=\"representative_name\">IL1A or IL1B</infon>\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P01583 or P01584</infon>\n",
      "<location length=\"4\" offset=\"1046\"></location>\n",
      "<text>IL-1</text>\n",
      "</annotation>\n",
      "<annotation id=\"A85\">\n",
      "<infon key=\"representative_name\">IL33</infon>\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">O95760</infon>\n",
      "<location length=\"5\" offset=\"1055\"></location>\n",
      "<text>IL-33</text>\n",
      "</annotation>\n",
      "<annotation id=\"A156\">\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<location length=\"11\" offset=\"1119\"></location>\n",
      "<text>coronavirus</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>32007643</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.7921</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>Preliminary estimation of the basic reproduction number of novel coronavirus (2019-nCoV) in China, from 2019 to 2020: A data-driven analysis in the early phase of the outbreak. </text>\n",
      "<annotation id=\"A104\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"17\" offset=\"59\"></location>\n",
      "<text>novel coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A105\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"78\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>177</offset>\n",
      "<text>BACKGROUNDS: An ongoing outbreak of a novel coronavirus (2019-nCoV) pneumonia hit a major city in China, Wuhan, December 2019 and subsequently reached other provinces/regions of China and other countries. We present estimates of the basic reproduction number, R0, of 2019-nCoV in the early phase of the outbreak. METHODS: Accounting for the impact of the variations in disease reporting rate, we modelled the epidemic curve of 2019-nCoV cases time series, in mainland China from January 10 to January 24, 2020, through the exponential growth. With the estimated intrinsic growth rate (γ), we estimated R0 by using the serial intervals (SI) of two other well-known coronavirus diseases, MERS and SARS, as approximations for the true unknown SI. FINDINGS: The early outbreak data largely follows the exponential growth. We estimated that the mean R0 ranges from 2.24 (95%CI: 1.96-2.55) to 3.58 (95%CI: 2.89-4.39) associated with 8-fold to 2-fold increase in the reporting rate. We demonstrated that changes in reporting rate substantially affect estimates of R0. CONCLUSION: The mean estimate of R0 for the 2019-nCoV ranges from 2.24 to 3.58, and is significantly larger than 1. Our findings indicate the potential of 2019-nCoV to cause outbreaks. </text>\n",
      "<annotation id=\"A106\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"39\" offset=\"215\"></location>\n",
      "<text>novel coronavirus (2019-nCoV) pneumonia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A107\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"444\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A108\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"604\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A109\">\n",
      "<infon key=\"representative_name\">coronavirus infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"20\" offset=\"841\"></location>\n",
      "<text>coronavirus diseases</text>\n",
      "</annotation>\n",
      "<annotation id=\"A110\">\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"4\" offset=\"863\"></location>\n",
      "<text>MERS</text>\n",
      "</annotation>\n",
      "<annotation id=\"A111\">\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"4\" offset=\"872\"></location>\n",
      "<text>SARS</text>\n",
      "</annotation>\n",
      "<annotation id=\"A112\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"1282\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A113\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"1393\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>32036774</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.7925</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>Emerging novel coronavirus (2019-nCoV)-current scenario, evolutionary perspective based on genome analysis and recent developments. </text>\n",
      "<annotation id=\"A114\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"26\" offset=\"0\"></location>\n",
      "<text>Emerging novel coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A115\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"28\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>132</offset>\n",
      "<text>Coronaviruses are the well-known cause of severe respiratory, enteric and systemic infections in a wide range of hosts including man, mammals, fish, and avian. The scientific interest on coronaviruses increased after the emergence of Severe Acute Respiratory Syndrome coronavirus (SARS-CoV) outbreaks in 2002-2003 followed by Middle East Respiratory Syndrome CoV (MERS-CoV). This decade's first CoV, named 2019-nCoV, emerged from Wuhan, China, and declared as 'Public Health Emergency of International Concern' on January 30th, 2020 by the World Health Organization (WHO). As on February 4, 2020, 425 deaths reported in China only and one death outside China (Philippines). In a short span of time, the virus spread has been noted in 24 countries. The zoonotic transmission (animal-to-human) is suspected as the route of disease origin. The genetic analyses predict bats as the most probable source of 2019-nCoV though further investigations needed to confirm the origin of the novel virus. The ongoing nCoV outbreak highlights the hidden wild animal reservoir of the deadly viruses and possible threat of spillover zoonoses as well. The successful virus isolation attempts have made doors open for developing better diagnostics and effective vaccines helping in combating the spread of the virus to newer areas. </text>\n",
      "<annotation id=\"A116\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"NCBI:txid\">693995</infon>\n",
      "<location length=\"13\" offset=\"132\"></location>\n",
      "<text>Coronaviruses</text>\n",
      "</annotation>\n",
      "<annotation id=\"A117\">\n",
      "<infon key=\"representative_name\">respiratory infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"11\" offset=\"181\"></location>\n",
      "<text>respiratory</text>\n",
      "</annotation>\n",
      "<annotation id=\"A118\">\n",
      "<infon key=\"representative_name\">enteric infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"7\" offset=\"194\"></location>\n",
      "<text>enteric</text>\n",
      "</annotation>\n",
      "<annotation id=\"A119\">\n",
      "<infon key=\"representative_name\">systemic infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"19\" offset=\"206\"></location>\n",
      "<text>systemic infections</text>\n",
      "</annotation>\n",
      "<annotation id=\"A120\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"NCBI:txid\">693995</infon>\n",
      "<location length=\"13\" offset=\"319\"></location>\n",
      "<text>coronaviruses</text>\n",
      "</annotation>\n",
      "<annotation id=\"A121\">\n",
      "<infon key=\"representative_name\">SARS-CoV</infon>\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<location length=\"45\" offset=\"366\"></location>\n",
      "<text>Severe Acute Respiratory Syndrome coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A122\">\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<location length=\"8\" offset=\"413\"></location>\n",
      "<text>SARS-CoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A123\">\n",
      "<infon key=\"representative_name\">MERS-CoV</infon>\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<location length=\"36\" offset=\"458\"></location>\n",
      "<text>Middle East Respiratory Syndrome CoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A124\">\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<infon key=\"NCBI:txid\">1335626</infon>\n",
      "<location length=\"8\" offset=\"496\"></location>\n",
      "<text>MERS-CoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A125\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"23\" offset=\"507\"></location>\n",
      "<text>This decade's first CoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A126\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"538\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A127\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"1034\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A128\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"4\" offset=\"1135\"></location>\n",
      "<text>nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A129\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"11\" offset=\"1110\"></location>\n",
      "<text>novel virus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A130\">\n",
      "<infon key=\"representative_name\">virus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<location length=\"7\" offset=\"1207\"></location>\n",
      "<text>viruses</text>\n",
      "</annotation>\n",
      "<annotation id=\"A131\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"5\" offset=\"1281\"></location>\n",
      "<text>virus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A132\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"5\" offset=\"1423\"></location>\n",
      "<text>virus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A133\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"5\" offset=\"835\"></location>\n",
      "<text>virus</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>31992388</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.8058</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>Real-time tentative assessment of the epidemiological characteristics of novel coronavirus infections in Wuhan, China, as at 22 January 2020. </text>\n",
      "<annotation id=\"A134\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"28\" offset=\"73\"></location>\n",
      "<text>novel coronavirus infections</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>142</offset>\n",
      "<text>A novel coronavirus (2019-nCoV) causing severe acute respiratory disease emerged recently in Wuhan, China. Information on reported cases strongly indicates human-to-human spread, and the most recent information is increasingly indicative of sustained human-to-human transmission. While the overall severity profile among cases may change as more mild cases are identified, we estimate a risk of fatality among hospitalised cases at 14% (95% confidence interval: 3.9-32%). </text>\n",
      "<annotation id=\"A135\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"17\" offset=\"144\"></location>\n",
      "<text>novel coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A136\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"32\" offset=\"182\"></location>\n",
      "<text>severe acute respiratory disease</text>\n",
      "</annotation>\n",
      "<annotation id=\"A137\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"163\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>32020836</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.8286</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>RNA based mNGS approach identifies a novel human coronavirus from two individual pneumonia cases in 2019 Wuhan outbreak. </text>\n",
      "<annotation id=\"A86\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"23\" offset=\"37\"></location>\n",
      "<text>novel human coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A87\">\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"9\" offset=\"81\"></location>\n",
      "<text>pneumonia</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>121</offset>\n",
      "<text>From December 2019, an outbreak of unusual pneumonia was reported in Wuhan with many cases linked to Huanan Seafood Market that sells seafood as well as live exotic animals. We investigated two patients who developed acute respiratory syndromes after independent contact history with this market. The two patients shared common clinical features including fever, cough, and multiple ground-glass opacities in the bilateral lung field with patchy infiltration. Here, we highlight the use of a low-input metagenomic next-generation sequencing (mNGS) approach on RNA extracted from bronchoalveolar lavage fluid (BALF). It rapidly identified a novel coronavirus (named 2019-nCoV according to World Health Organization announcement) which was the sole pathogens in the sample with very high abundance level (1.5% and 0.62% of total RNA sequenced). The entire viral genome is 29,881 nt in length (GenBank MN988668 and MN988669, Sequence Read Archive database Bioproject accession PRJNA601736) and is classified into β-coronavirus genus. Phylogenetic analysis indicates that 2019-nCoV is close to coronaviruses (CoVs) circulating in Rhinolophus (Horseshoe bats), such as 98.7% nucleotide identity to partial RdRp gene of bat coronavirus strain BtCoV/4991 (GenBank KP876546, 370 nt sequence of RdRp and lack of other genome sequence) and 87.9% nucleotide identity to bat coronavirus strain bat-SL-CoVZC45 and bat-SL-CoVZXC21. Evolutionary analysis based on ORF1a/1b, S, and N genes also suggests 2019-nCoV is more likely a novel CoV independently introduced from animals to humans. </text>\n",
      "<annotation id=\"A88\">\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"9\" offset=\"164\"></location>\n",
      "<text>pneumonia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A89\">\n",
      "<infon key=\"representative_name\">respiratory disease</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"21\" offset=\"344\"></location>\n",
      "<text>respiratory syndromes</text>\n",
      "</annotation>\n",
      "<annotation id=\"A90\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"477\"></location>\n",
      "<text>fever</text>\n",
      "</annotation>\n",
      "<annotation id=\"A91\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"484\"></location>\n",
      "<text>cough</text>\n",
      "</annotation>\n",
      "<annotation id=\"A92\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"17\" offset=\"761\"></location>\n",
      "<text>novel coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A93\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"786\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A94\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<location length=\"13\" offset=\"1131\"></location>\n",
      "<text>β-coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A95\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"1189\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A96\">\n",
      "<infon key=\"representative_name\">bat coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<location length=\"13\" offset=\"1211\"></location>\n",
      "<text>coronaviruses</text>\n",
      "</annotation>\n",
      "<annotation id=\"A97\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<location length=\"4\" offset=\"1226\"></location>\n",
      "<text>CoVs</text>\n",
      "</annotation>\n",
      "<annotation id=\"A98\">\n",
      "<infon key=\"representative_name\">Rhinolophus bat coronavirus BtCoV/4991</infon>\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<infon key=\"NCBI:txid\">1788497</infon>\n",
      "<location length=\"33\" offset=\"1335\"></location>\n",
      "<text>bat coronavirus strain BtCoV/4991</text>\n",
      "</annotation>\n",
      "<annotation id=\"A99\">\n",
      "<infon key=\"representative_name\">bat coronavirus</infon>\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<location length=\"37\" offset=\"1480\"></location>\n",
      "<text>bat coronavirus strain bat-SL-CoVZC45</text>\n",
      "</annotation>\n",
      "<annotation id=\"A100\">\n",
      "<infon key=\"representative_name\">bat coronavirus</infon>\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<location length=\"15\" offset=\"1522\"></location>\n",
      "<text>bat-SL-CoVZXC21</text>\n",
      "</annotation>\n",
      "<annotation id=\"A101\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"1609\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A102\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<location length=\"3\" offset=\"1642\"></location>\n",
      "<text>CoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A103\">\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">A0A126KB67</infon>\n",
      "<location length=\"4\" offset=\"1407\"></location>\n",
      "<text>RdRp</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>32015508</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.8307</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>A new coronavirus associated with human respiratory disease in China. </text>\n",
      "<annotation id=\"A138\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"15\" offset=\"2\"></location>\n",
      "<text>new coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A139\">\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"19\" offset=\"40\"></location>\n",
      "<text>respiratory disease</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>70</offset>\n",
      "<text>Emerging infectious diseases, such as severe acute respiratory syndrome (SARS) and Zika virus disease, present a major threat to public health1-3. Despite intense research efforts, how, when and where new diseases appear are still a source of considerable uncertainty. A severe respiratory disease was recently reported in Wuhan, Hubei province, China. As of 25 January 2020, at least 1,975 cases had been reported since the first patient was hospitalized on 12 December 2019. Epidemiological investigations have suggested that the outbreak was associated with a seafood market in Wuhan. Here we study a single patient who was a worker at the market and who was admitted to the Central Hospital of Wuhan on 26 December 2019 while experiencing a severe respiratory syndrome that included fever, dizziness and a cough. Metagenomic RNA sequencing4 of a sample of bronchoalveolar lavage fluid from the patient identified a new RNA virus strain from the family Coronaviridae, which is designated here 'WH-Human 1' coronavirus (and has also been referred to as '2019-nCoV'). Phylogenetic analysis of the complete viral genome (29,903 nucleotides) revealed that the virus was most closely related (89.1% nucleotide similarity) to a group of SARS-like coronaviruses (genus Betacoronavirus, subgenus Sarbecovirus) that had previously been found in bats in China5. This outbreak highlights the ongoing ability of viral spill-over from animals to cause severe disease in humans. </text>\n",
      "<annotation id=\"A140\">\n",
      "<infon key=\"representative_name\">infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"19\" offset=\"79\"></location>\n",
      "<text>infectious diseases</text>\n",
      "</annotation>\n",
      "<annotation id=\"A141\">\n",
      "<infon key=\"representative_name\">SARS</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"33\" offset=\"108\"></location>\n",
      "<text>severe acute respiratory syndrome</text>\n",
      "</annotation>\n",
      "<annotation id=\"A142\">\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"4\" offset=\"143\"></location>\n",
      "<text>SARS</text>\n",
      "</annotation>\n",
      "<annotation id=\"A143\">\n",
      "<infon key=\"representative_name\">Zika virus infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"18\" offset=\"153\"></location>\n",
      "<text>Zika virus disease</text>\n",
      "</annotation>\n",
      "<annotation id=\"A144\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"26\" offset=\"341\"></location>\n",
      "<text>severe respiratory disease</text>\n",
      "</annotation>\n",
      "<annotation id=\"A145\">\n",
      "<infon key=\"representative_name\">respiratory disease</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"20\" offset=\"822\"></location>\n",
      "<text>respiratory syndrome</text>\n",
      "</annotation>\n",
      "<annotation id=\"A146\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"857\"></location>\n",
      "<text>fever</text>\n",
      "</annotation>\n",
      "<annotation id=\"A147\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"9\" offset=\"864\"></location>\n",
      "<text>dizziness</text>\n",
      "</annotation>\n",
      "<annotation id=\"A148\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"880\"></location>\n",
      "<text>cough</text>\n",
      "</annotation>\n",
      "<annotation id=\"A149\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"50\" offset=\"989\"></location>\n",
      "<text>new RNA virus strain from the family Coronaviridae</text>\n",
      "</annotation>\n",
      "<annotation id=\"A150\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"24\" offset=\"1066\"></location>\n",
      "<text>'WH-Human 1' coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A151\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"1126\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A152\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"5\" offset=\"1229\"></location>\n",
      "<text>virus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A153\">\n",
      "<infon key=\"representative_name\">SARS-related coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"ncbitax:\">694009</infon>\n",
      "<location length=\"23\" offset=\"1304\"></location>\n",
      "<text>SARS-like coronaviruses</text>\n",
      "</annotation>\n",
      "<annotation id=\"A154\">\n",
      "<infon key=\"representative_name\">betacoronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"ncbitax:\">694002</infon>\n",
      "<location length=\"15\" offset=\"1335\"></location>\n",
      "<text>Betacoronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A155\">\n",
      "<infon key=\"representative_name\">sarbecovirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"ncbitax:\">2509511</infon>\n",
      "<location length=\"12\" offset=\"1361\"></location>\n",
      "<text>Sarbecovirus</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>31986264</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.8457</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>Clinical features of patients infected with 2019 novel coronavirus in Wuhan, China. </text>\n",
      "<annotation id=\"A10\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<location length=\"22\" offset=\"44\"></location>\n",
      "<text>2019 novel coronavirus</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>84</offset>\n",
      "<text>BACKGROUND: A recent cluster of pneumonia cases in Wuhan, China, was caused by a novel betacoronavirus, the 2019 novel coronavirus (2019-nCoV). We report the epidemiological, clinical, laboratory, and radiological characteristics and treatment and clinical outcomes of these patients. METHODS: All patients with suspected 2019-nCoV were admitted to a designated hospital in Wuhan. We prospectively collected and analysed data on patients with laboratory-confirmed 2019-nCoV infection by real-time RT-PCR and next-generation sequencing. Data were obtained with standardised data collection forms shared by WHO and the International Severe Acute Respiratory and Emerging Infection Consortium from electronic medical records. Researchers also directly communicated with patients or their families to ascertain epidemiological and symptom data. Outcomes were also compared between patients who had been admitted to the intensive care unit (ICU) and those who had not. FINDINGS: By Jan 2, 2020, 41 admitted hospital patients had been identified as having laboratory-confirmed 2019-nCoV infection. Most of the infected patients were men (30 [73%] of 41); less than half had underlying diseases (13 [32%]), including diabetes (eight [20%]), hypertension (six [15%]), and cardiovascular disease (six [15%]). Median age was 49·0 years (IQR 41·0-58·0). 27 (66%) of 41 patients had been exposed to Huanan seafood market. One family cluster was found. Common symptoms at onset of illness were fever (40 [98%] of 41 patients), cough (31 [76%]), and myalgia or fatigue (18 [44%]); less common symptoms were sputum production (11 [28%] of 39), headache (three [8%] of 38), haemoptysis (two [5%] of 39), and diarrhoea (one [3%] of 38). Dyspnoea developed in 22 (55%) of 40 patients (median time from illness onset to dyspnoea 8·0 days [IQR 5·0-13·0]). 26 (63%) of 41 patients had lymphopenia. All 41 patients had pneumonia with abnormal findings on chest CT. Complications included acute respiratory distress syndrome (12 [29%]), RNAaemia (six [15%]), acute cardiac injury (five [12%]) and secondary infection (four [10%]). 13 (32%) patients were admitted to an ICU and six (15%) died. Compared with non-ICU patients, ICU patients had higher plasma levels of IL2, IL7, IL10, GSCF, IP10, MCP1, MIP1A, and TNFα. INTERPRETATION: The 2019-nCoV infection caused clusters of severe respiratory illness similar to severe acute respiratory syndrome coronavirus and was associated with ICU admission and high mortality. Major gaps in our knowledge of the origin, epidemiology, duration of human transmission, and clinical spectrum of disease need fulfilment by future studies. FUNDING: Ministry of Science and Technology, Chinese Academy of Medical Sciences, National Natural Science Foundation of China, and Beijing Municipal Science and Technology Commission. </text>\n",
      "<annotation id=\"A7\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"9\" offset=\"116\"></location>\n",
      "<text>pneumonia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A8\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<location length=\"21\" offset=\"165\"></location>\n",
      "<text>novel betacoronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A9\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<location length=\"22\" offset=\"192\"></location>\n",
      "<text>2019 novel coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A11\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<location length=\"9\" offset=\"216\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A12\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<location length=\"9\" offset=\"406\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A13\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"19\" offset=\"548\"></location>\n",
      "<text>2019-nCoV infection</text>\n",
      "</annotation>\n",
      "<annotation id=\"A14\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"19\" offset=\"1155\"></location>\n",
      "<text>2019-nCoV infection</text>\n",
      "</annotation>\n",
      "<annotation id=\"A15\">\n",
      "<infon key=\"representative_name\">diabetes mellitus</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"8\" offset=\"1294\"></location>\n",
      "<text>diabetes</text>\n",
      "</annotation>\n",
      "<annotation id=\"A16\">\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"12\" offset=\"1318\"></location>\n",
      "<text>hypertension</text>\n",
      "</annotation>\n",
      "<annotation id=\"A17\">\n",
      "<infon key=\"representative_name\">cardiovascular disease</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"22\" offset=\"1348\"></location>\n",
      "<text>cardiovascular disease</text>\n",
      "</annotation>\n",
      "<annotation id=\"A18\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"1565\"></location>\n",
      "<text>fever</text>\n",
      "</annotation>\n",
      "<annotation id=\"A19\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"5\" offset=\"1598\"></location>\n",
      "<text>cough</text>\n",
      "</annotation>\n",
      "<annotation id=\"A20\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"7\" offset=\"1620\"></location>\n",
      "<text>myalgia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A21\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"7\" offset=\"1631\"></location>\n",
      "<text>fatigue</text>\n",
      "</annotation>\n",
      "<annotation id=\"A22\">\n",
      "<infon key=\"representative_name\">sputum production</infon>\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"17\" offset=\"1677\"></location>\n",
      "<text>sputum production</text>\n",
      "</annotation>\n",
      "<annotation id=\"A23\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"8\" offset=\"1713\"></location>\n",
      "<text>headache</text>\n",
      "</annotation>\n",
      "<annotation id=\"A24\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"11\" offset=\"1742\"></location>\n",
      "<text>haemoptysis</text>\n",
      "</annotation>\n",
      "<annotation id=\"A26\">\n",
      "<infon key=\"representative_name\">dairrhoea</infon>\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"9\" offset=\"1776\"></location>\n",
      "<text>diarrhoea</text>\n",
      "</annotation>\n",
      "<annotation id=\"A27\">\n",
      "<infon key=\"representative_name\">dyspnoea</infon>\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"8\" offset=\"1804\"></location>\n",
      "<text>Dyspnoea</text>\n",
      "</annotation>\n",
      "<annotation id=\"A28\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"8\" offset=\"1885\"></location>\n",
      "<text>dyspnoea</text>\n",
      "</annotation>\n",
      "<annotation id=\"A29\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"11\" offset=\"1948\"></location>\n",
      "<text>lymphopenia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A30\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"9\" offset=\"1981\"></location>\n",
      "<text>pneumonia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A31\">\n",
      "<infon key=\"representative_name\">acute respiratory distress syndrome</infon>\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"35\" offset=\"2050\"></location>\n",
      "<text>acute respiratory distress syndrome</text>\n",
      "</annotation>\n",
      "<annotation id=\"A32\">\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"8\" offset=\"2098\"></location>\n",
      "<text>RNAaemia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A33\">\n",
      "<infon key=\"representative_name\">acute cardiac injury</infon>\n",
      "<infon key=\"type\">Symptom</infon>\n",
      "<location length=\"20\" offset=\"2120\"></location>\n",
      "<text>acute cardiac injury</text>\n",
      "</annotation>\n",
      "<annotation id=\"A34\">\n",
      "<infon key=\"representative_name\">secondary infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"19\" offset=\"2158\"></location>\n",
      "<text>secondary infection</text>\n",
      "</annotation>\n",
      "<annotation id=\"A35\">\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P60568</infon>\n",
      "<location length=\"3\" offset=\"2327\"></location>\n",
      "<text>IL2</text>\n",
      "</annotation>\n",
      "<annotation id=\"A36\">\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P13232</infon>\n",
      "<location length=\"3\" offset=\"2332\"></location>\n",
      "<text>IL7</text>\n",
      "</annotation>\n",
      "<annotation id=\"A37\">\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P22301</infon>\n",
      "<location length=\"4\" offset=\"2337\"></location>\n",
      "<text>IL10</text>\n",
      "</annotation>\n",
      "<annotation id=\"A38\">\n",
      "<infon key=\"representative_name\">CSF3</infon>\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P09919</infon>\n",
      "<location length=\"4\" offset=\"2343\"></location>\n",
      "<text>GSCF</text>\n",
      "</annotation>\n",
      "<annotation id=\"A39\">\n",
      "<infon key=\"representative_name\">CXCL10</infon>\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P02778</infon>\n",
      "<location length=\"4\" offset=\"2349\"></location>\n",
      "<text>IP10</text>\n",
      "</annotation>\n",
      "<annotation id=\"A40\">\n",
      "<infon key=\"representative_name\">CCL2</infon>\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P13500</infon>\n",
      "<location length=\"4\" offset=\"2355\"></location>\n",
      "<text>MCP1</text>\n",
      "</annotation>\n",
      "<annotation id=\"A41\">\n",
      "<infon key=\"representative_name\">CCL3</infon>\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P10147</infon>\n",
      "<location length=\"5\" offset=\"2361\"></location>\n",
      "<text>MIP1A</text>\n",
      "</annotation>\n",
      "<annotation id=\"A42\">\n",
      "<infon key=\"representative_name\">TNF</infon>\n",
      "<infon key=\"type\">Protein</infon>\n",
      "<infon key=\"Uniprot\">P01375</infon>\n",
      "<location length=\"4\" offset=\"2372\"></location>\n",
      "<text>TNFα</text>\n",
      "</annotation>\n",
      "<annotation id=\"A43\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"19\" offset=\"2398\"></location>\n",
      "<text>2019-nCoV infection</text>\n",
      "</annotation>\n",
      "<annotation id=\"A44\">\n",
      "<infon key=\"representative_name\">severe respiratory disease</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"26\" offset=\"2437\"></location>\n",
      "<text>severe respiratory illness</text>\n",
      "</annotation>\n",
      "<annotation id=\"A45\">\n",
      "<infon key=\"representative_name\">SARS-CoV</infon>\n",
      "<infon key=\"type\">Virus_other</infon>\n",
      "<location length=\"45\" offset=\"2475\"></location>\n",
      "<text>severe acute respiratory syndrome coronavirus</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>32029004</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.8626</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>A rapid advice guideline for the diagnosis and treatment of 2019 novel coronavirus (2019-nCoV) infected pneumonia (standard version). </text>\n",
      "<annotation id=\"A46\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"53\" offset=\"60\"></location>\n",
      "<text>2019 novel coronavirus (2019-nCoV) infected pneumonia</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>134</offset>\n",
      "<text>In December 2019, a new type viral pneumonia cases occurred in Wuhan, Hubei Province; and then named \"2019 novel coronavirus (2019-nCoV)\" by the World Health Organization (WHO) on 12 January 2020. For it is a never been experienced respiratory disease before and with infection ability widely and quickly, it attracted the world's attention but without treatment and control manual. For the request from frontline clinicians and public health professionals of 2019-nCoV infected pneumonia management, an evidence-based guideline urgently needs to be developed. Therefore, we drafted this guideline according to the rapid advice guidelines methodology and general rules of WHO guideline development; we also added the first-hand management data of Zhongnan Hospital of Wuhan University. This guideline includes the guideline methodology, epidemiological characteristics, disease screening and population prevention, diagnosis, treatment and control (including traditional Chinese Medicine), nosocomial infection prevention and control, and disease nursing of the 2019-nCoV. Moreover, we also provide a whole process of a successful treatment case of the severe 2019-nCoV infected pneumonia and experience and lessons of hospital rescue for 2019-nCoV infections. This rapid advice guideline is suitable for the first frontline doctors and nurses, managers of hospitals and healthcare sections, community residents, public health persons, relevant researchers, and all person who are interested in the 2019-nCoV. </text>\n",
      "<annotation id=\"A47\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"24\" offset=\"154\"></location>\n",
      "<text>new type viral pneumonia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A48\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<location length=\"22\" offset=\"236\"></location>\n",
      "<text>2019 novel coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A49\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<location length=\"9\" offset=\"260\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A50\">\n",
      "<infon key=\"representative_name\">respiratory disease</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"19\" offset=\"366\"></location>\n",
      "<text>respiratory disease</text>\n",
      "</annotation>\n",
      "<annotation id=\"A51\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"28\" offset=\"594\"></location>\n",
      "<text>2019-nCoV infected pneumonia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A52\">\n",
      "<infon key=\"representative_name\">nosocomial infection</infon>\n",
      "<infon key=\"type\">Disease_other</infon>\n",
      "<location length=\"20\" offset=\"1124\"></location>\n",
      "<text>nosocomial infection</text>\n",
      "</annotation>\n",
      "<annotation id=\"A53\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"9\" offset=\"1196\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A54\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"28\" offset=\"1294\"></location>\n",
      "<text>2019-nCoV infected pneumonia</text>\n",
      "</annotation>\n",
      "<annotation id=\"A55\">\n",
      "<infon key=\"representative_name\">COVID-19</infon>\n",
      "<infon key=\"type\">Disease_COVID-19</infon>\n",
      "<location length=\"20\" offset=\"1373\"></location>\n",
      "<text>2019-nCoV infections</text>\n",
      "</annotation>\n",
      "<annotation id=\"A56\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<location length=\"9\" offset=\"1633\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "<document>\n",
      "<id>31996494</id>\n",
      "<infon key=\"BQ_FROM\">PubMed</infon>\n",
      "<infon key=\"BQ_PIESCORE\">-0.8683</infon>\n",
      "<infon key=\"BQ_DONE\">NO</infon>\n",
      "<infon key=\"BQ_CURATABLE\">YES</infon>\n",
      "<passage>\n",
      "<infon key=\"type\">title</infon>\n",
      "<offset>0</offset>\n",
      "<text>Drug treatment options for the 2019-new coronavirus (2019-nCoV). </text>\n",
      "<annotation id=\"A55\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"20\" offset=\"31\"></location>\n",
      "<text>2019-new coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A56\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"53\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "<passage>\n",
      "<infon key=\"type\">abstract</infon>\n",
      "<offset>65</offset>\n",
      "<text>As of January 22, 2020, a total of 571 cases of the 2019-new coronavirus (2019-nCoV) have been reported in 25 provinces (districts and cities) in China. At present, there is no vaccine or antiviral treatment for human and animal coronavirus, so that identifying the drug treatment options as soon as possible is critical for the response to the 2019-nCoV outbreak. Three general methods, which include existing broad-spectrum antiviral drugs using standard assays, screening of a chemical library containing many existing compounds or databases, and the redevelopment of new specific drugs based on the genome and biophysical understanding of individual coronaviruses, are used to discover the potential antiviral treatment of human pathogen coronavirus. Lopinavir /Ritonavir, Nucleoside analogues, Neuraminidase inhibitors, Remdesivir, peptide (EK1), arbidol, RNA synthesis inhibitors (such as TDF, 3TC), anti-inflammatory drugs (such as hormones and other molecules), Chinese traditional medicine, such ShuFengJieDu Capsules and Lianhuaqingwen Capsule, could be the drug treatment options for 2019-nCoV. However, the efficacy and safety of these drugs for 2019- nCoV still need to be further confirmed by clinical experiments. </text>\n",
      "<annotation id=\"A57\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"20\" offset=\"117\"></location>\n",
      "<text>2019-new coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A58\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"139\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A59\">\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"NCBI:txid\">693995</infon>\n",
      "<location length=\"11\" offset=\"294\"></location>\n",
      "<text>coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A60\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"410\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A61\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<infon key=\"NCBI:txid\">693995</infon>\n",
      "<location length=\"13\" offset=\"719\"></location>\n",
      "<text>coronaviruses</text>\n",
      "</annotation>\n",
      "<annotation id=\"A62\">\n",
      "<infon key=\"representative_name\">coronavirus</infon>\n",
      "<infon key=\"type\">Virus_family</infon>\n",
      "<location length=\"26\" offset=\"792\"></location>\n",
      "<text>human pathogen coronavirus</text>\n",
      "</annotation>\n",
      "<annotation id=\"A63\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"9\" offset=\"1160\"></location>\n",
      "<text>2019-nCoV</text>\n",
      "</annotation>\n",
      "<annotation id=\"A64\">\n",
      "<infon key=\"representative_name\">SARS-CoV-2</infon>\n",
      "<infon key=\"type\">Virus_SARS-CoV-2</infon>\n",
      "<infon key=\"NCBI:txid\">2697049</infon>\n",
      "<location length=\"10\" offset=\"1223\"></location>\n",
      "<text>2019- nCoV</text>\n",
      "</annotation>\n",
      "</passage>\n",
      "</document>\n",
      "</collection>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import collections\n",
    "import csv\n",
    "import sys\n",
    "import codecs\n",
    "from xml.dom.minidom import parse\n",
    "import xml.dom.minidom\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "\n",
    "with codecs.open(\"../corona/manuscript/Supplemental_file4.xml\", \"r\", \"utf-8\") as file:\n",
    "    soup = BeautifulSoup(file, \"html.parser\")\n",
    "print(soup)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An IOB2 format all the tokens must be tagged with \"B\", \"I\", or \"O\". If the tokens are name entities they could be B or I, else O. All the punctuation signs are tagged by \"O\"."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Return O\n",
      "of O\n",
      "the O\n",
      "Coronavirus B-Virus_family\n",
      ": O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "\n",
      "\n",
      "The O\n",
      "emergence O\n",
      "of O\n",
      "a O\n",
      "novel B-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "has O\n",
      "awakened O\n",
      "the O\n",
      "echoes O\n",
      "of O\n",
      "SARS-CoV B-Virus_other\n",
      "from O\n",
      "nearly O\n",
      "two O\n",
      "decades O\n",
      "ago O\n",
      "\n",
      "\n",
      "Yet O\n",
      ", O\n",
      "with O\n",
      "technological O\n",
      "advances O\n",
      "and O\n",
      "important O\n",
      "lessons O\n",
      "gained O\n",
      "from O\n",
      "previous O\n",
      "outbreaks O\n",
      ", O\n",
      "perhaps O\n",
      "the O\n",
      "world O\n",
      "is O\n",
      "better O\n",
      "equipped O\n",
      "to O\n",
      "deal O\n",
      "with O\n",
      "the O\n",
      "most B-Virus_SARS-CoV-2\n",
      "recent I-Virus_SARS-CoV-2\n",
      "emergent I-Virus_SARS-CoV-2\n",
      "group I-Virus_SARS-CoV-2\n",
      "2B I-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "\n",
      "\n",
      "Mast O\n",
      "cells O\n",
      "contribute O\n",
      "to O\n",
      "coronavirus-induced O\n",
      "inflammation B-Symptom\n",
      ": O\n",
      "new O\n",
      "anti-inflammatory O\n",
      "strategy O\n",
      "\n",
      "\n",
      "Coronavirus B-Virus_SARS-CoV-2\n",
      ", O\n",
      "which O\n",
      "can O\n",
      "cause O\n",
      "respiratory B-Disease_other\n",
      "syndrome I-Disease_other\n",
      ", O\n",
      "to O\n",
      "date O\n",
      "has O\n",
      "affected O\n",
      "over O\n",
      "seventeen O\n",
      "thousand O\n",
      "individuals O\n",
      ", O\n",
      "especially O\n",
      "in O\n",
      "China O\n",
      "\n",
      "\n",
      "Coronavirus B-Virus_SARS-CoV-2\n",
      "is O\n",
      "interspecies O\n",
      "and O\n",
      "can O\n",
      "also O\n",
      "be O\n",
      "transmitted O\n",
      "from O\n",
      "man O\n",
      "to O\n",
      "man O\n",
      ", O\n",
      "with O\n",
      "an O\n",
      "incubation O\n",
      "ranging O\n",
      "from O\n",
      "1 O\n",
      "to O\n",
      "14 O\n",
      "days O\n",
      "\n",
      "\n",
      "Human B-Disease_other\n",
      "coronavirus I-Disease_other\n",
      "infections I-Disease_other\n",
      "can O\n",
      "induce O\n",
      "not O\n",
      "only O\n",
      "mild O\n",
      "to O\n",
      "severe O\n",
      "respiratory B-Disease_other\n",
      "diseases I-Disease_other\n",
      ", O\n",
      "but O\n",
      "also O\n",
      "inflammation B-Symptom\n",
      ", O\n",
      "high O\n",
      "fever B-Symptom\n",
      ", O\n",
      "cough B-Symptom\n",
      ", O\n",
      "acute O\n",
      "respiratory B-Disease_other\n",
      "tract I-Disease_other\n",
      "infection I-Disease_other\n",
      "and O\n",
      "dysfunction B-Symptom\n",
      "of I-Symptom\n",
      "internal I-Symptom\n",
      "organs I-Symptom\n",
      "that O\n",
      "may O\n",
      "lead O\n",
      "to O\n",
      "death B-Symptom\n",
      "\n",
      "\n",
      "Coronavirus B-Virus_SARS-CoV-2\n",
      "infection O\n",
      "( O\n",
      "regardless O\n",
      "of O\n",
      "the O\n",
      "various O\n",
      "types O\n",
      "of O\n",
      "corona B-Virus_family\n",
      "virus I-Virus_family\n",
      ") O\n",
      "is O\n",
      "primarily O\n",
      "attacked O\n",
      "by O\n",
      "immune O\n",
      "cells O\n",
      "including O\n",
      "mast O\n",
      "cells O\n",
      "( O\n",
      "MCs O\n",
      ") O\n",
      ", O\n",
      "which O\n",
      "are O\n",
      "located O\n",
      "in O\n",
      "the O\n",
      "submucosa O\n",
      "of O\n",
      "the O\n",
      "respiratory O\n",
      "tract O\n",
      "and O\n",
      "in O\n",
      "the O\n",
      "nasal O\n",
      "cavity O\n",
      "and O\n",
      "represent O\n",
      "a O\n",
      "barrier O\n",
      "of O\n",
      "protection O\n",
      "against O\n",
      "microorganisms O\n",
      "\n",
      "\n",
      "Virus B-Virus_family\n",
      "activate O\n",
      "MCs O\n",
      "which O\n",
      "release O\n",
      "early O\n",
      "inflammatory O\n",
      "chemical O\n",
      "compounds O\n",
      "including O\n",
      "histamine O\n",
      "and O\n",
      "protease O\n",
      "; O\n",
      "while O\n",
      "late O\n",
      "activation O\n",
      "provokes O\n",
      "the O\n",
      "generation O\n",
      "of O\n",
      "pro-inflammatory O\n",
      "IL-1 B-Protein\n",
      "family O\n",
      "members O\n",
      "including O\n",
      "IL-1 B-Protein\n",
      "and O\n",
      "IL-33 B-Protein\n",
      "\n",
      "\n",
      "Here O\n",
      ", O\n",
      "we O\n",
      "propose O\n",
      "for O\n",
      "the O\n",
      "first O\n",
      "time O\n",
      "that O\n",
      "inflammation B-Symptom\n",
      "by O\n",
      "coronavirus B-Virus_family\n",
      "may O\n",
      "be O\n",
      "inhibited O\n",
      "by O\n",
      "anti-inflammatory O\n",
      "cytokines O\n",
      "belonging O\n",
      "to O\n",
      "the O\n",
      "IL-1 B-Protein\n",
      "family O\n",
      "members O\n",
      "\n",
      "\n",
      "Preliminary O\n",
      "estimation O\n",
      "of O\n",
      "the O\n",
      "basic O\n",
      "reproduction O\n",
      "number O\n",
      "of O\n",
      "novel B-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "in O\n",
      "China O\n",
      ", O\n",
      "from O\n",
      "2019 O\n",
      "to O\n",
      "2020 O\n",
      ": O\n",
      "A O\n",
      "data-driven O\n",
      "analysis O\n",
      "in O\n",
      "the O\n",
      "early O\n",
      "phase O\n",
      "of O\n",
      "the O\n",
      "outbreak O\n",
      "\n",
      "\n",
      "BACKGROUNDS O\n",
      ": O\n",
      "An O\n",
      "ongoing O\n",
      "outbreak O\n",
      "of O\n",
      "a O\n",
      "novel O\n",
      "coronavirus O\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "pneumonia O\n",
      "hit O\n",
      "a O\n",
      "major O\n",
      "city O\n",
      "in O\n",
      "China O\n",
      ", O\n",
      "Wuhan O\n",
      ", O\n",
      "December O\n",
      "2019 O\n",
      "and O\n",
      "subsequently O\n",
      "reached O\n",
      "other O\n",
      "provinces/regions O\n",
      "of O\n",
      "China O\n",
      "and O\n",
      "other O\n",
      "countries O\n",
      "\n",
      "\n",
      "We O\n",
      "present O\n",
      "estimates O\n",
      "of O\n",
      "the O\n",
      "basic O\n",
      "reproduction O\n",
      "number O\n",
      ", O\n",
      "R0 O\n",
      ", O\n",
      "of O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "in O\n",
      "the O\n",
      "early O\n",
      "phase O\n",
      "of O\n",
      "the O\n",
      "outbreak O\n",
      "\n",
      "\n",
      "METHODS O\n",
      ": O\n",
      "Accounting O\n",
      "for O\n",
      "the O\n",
      "impact O\n",
      "of O\n",
      "the O\n",
      "variations O\n",
      "in O\n",
      "disease O\n",
      "reporting O\n",
      "rate O\n",
      ", O\n",
      "we O\n",
      "modelled O\n",
      "the O\n",
      "epidemic O\n",
      "curve O\n",
      "of O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "cases O\n",
      "time O\n",
      "series O\n",
      ", O\n",
      "in O\n",
      "mainland O\n",
      "China O\n",
      "from O\n",
      "January O\n",
      "10 O\n",
      "to O\n",
      "January O\n",
      "24 O\n",
      ", O\n",
      "2020 O\n",
      ", O\n",
      "through O\n",
      "the O\n",
      "exponential O\n",
      "growth O\n",
      "\n",
      "\n",
      "With O\n",
      "the O\n",
      "estimated O\n",
      "intrinsic O\n",
      "growth O\n",
      "rate O\n",
      "( O\n",
      "γ O\n",
      ") O\n",
      ", O\n",
      "we O\n",
      "estimated O\n",
      "R0 O\n",
      "by O\n",
      "using O\n",
      "the O\n",
      "serial O\n",
      "intervals O\n",
      "( O\n",
      "SI O\n",
      ") O\n",
      "of O\n",
      "two O\n",
      "other O\n",
      "well-known O\n",
      "coronavirus B-Disease_other\n",
      "diseases I-Disease_other\n",
      ", O\n",
      "MERS B-Disease_other\n",
      "and O\n",
      "SARS B-Disease_other\n",
      ", O\n",
      "as O\n",
      "approximations O\n",
      "for O\n",
      "the O\n",
      "true O\n",
      "unknown O\n",
      "SI O\n",
      "\n",
      "\n",
      "FINDINGS O\n",
      ": O\n",
      "The O\n",
      "early O\n",
      "outbreak O\n",
      "data O\n",
      "largely O\n",
      "follows O\n",
      "the O\n",
      "exponential O\n",
      "growth O\n",
      "\n",
      "\n",
      "We O\n",
      "estimated O\n",
      "that O\n",
      "the O\n",
      "mean O\n",
      "R0 O\n",
      "ranges O\n",
      "from O\n",
      "2 O\n",
      "\n",
      "\n",
      "24 O\n",
      "( O\n",
      "95 O\n",
      "% O\n",
      "CI O\n",
      ": O\n",
      "1 O\n",
      "\n",
      "\n",
      "96-2 O\n",
      "\n",
      "\n",
      "55 O\n",
      ") O\n",
      "to O\n",
      "3 O\n",
      "\n",
      "\n",
      "58 O\n",
      "( O\n",
      "95 O\n",
      "% O\n",
      "CI O\n",
      ": O\n",
      "2 O\n",
      "\n",
      "\n",
      "89-4 O\n",
      "\n",
      "\n",
      "39 O\n",
      ") O\n",
      "associated O\n",
      "with O\n",
      "8-fold O\n",
      "to O\n",
      "2-fold O\n",
      "increase O\n",
      "in O\n",
      "the O\n",
      "reporting O\n",
      "rate O\n",
      "\n",
      "\n",
      "We O\n",
      "demonstrated O\n",
      "that O\n",
      "changes O\n",
      "in O\n",
      "reporting O\n",
      "rate O\n",
      "substantially O\n",
      "affect O\n",
      "estimates O\n",
      "of O\n",
      "R0 O\n",
      "\n",
      "\n",
      "CONCLUSION O\n",
      ": O\n",
      "The O\n",
      "mean O\n",
      "estimate O\n",
      "of O\n",
      "R0 O\n",
      "for O\n",
      "the O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "ranges O\n",
      "from O\n",
      "2 O\n",
      "\n",
      "\n",
      "24 O\n",
      "to O\n",
      "3 O\n",
      "\n",
      "\n",
      "58 O\n",
      ", O\n",
      "and O\n",
      "is O\n",
      "significantly O\n",
      "larger O\n",
      "than O\n",
      "1 O\n",
      "\n",
      "\n",
      "Our O\n",
      "findings O\n",
      "indicate O\n",
      "the O\n",
      "potential O\n",
      "of O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "to O\n",
      "cause O\n",
      "outbreaks O\n",
      "\n",
      "\n",
      "Emerging B-Virus_SARS-CoV-2\n",
      "novel I-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "-current O\n",
      "scenario O\n",
      ", O\n",
      "evolutionary O\n",
      "perspective O\n",
      "based O\n",
      "on O\n",
      "genome O\n",
      "analysis O\n",
      "and O\n",
      "recent O\n",
      "developments O\n",
      "\n",
      "\n",
      "Coronaviruses B-Virus_family\n",
      "are O\n",
      "the O\n",
      "well-known O\n",
      "cause O\n",
      "of O\n",
      "severe O\n",
      "respiratory B-Disease_other\n",
      ", O\n",
      "enteric B-Disease_other\n",
      "and O\n",
      "systemic B-Disease_other\n",
      "infections I-Disease_other\n",
      "in O\n",
      "a O\n",
      "wide O\n",
      "range O\n",
      "of O\n",
      "hosts O\n",
      "including O\n",
      "man O\n",
      ", O\n",
      "mammals O\n",
      ", O\n",
      "fish O\n",
      ", O\n",
      "and O\n",
      "avian O\n",
      "\n",
      "\n",
      "The O\n",
      "scientific O\n",
      "interest O\n",
      "on O\n",
      "coronaviruses B-Virus_family\n",
      "increased O\n",
      "after O\n",
      "the O\n",
      "emergence O\n",
      "of O\n",
      "Severe B-Virus_other\n",
      "Acute I-Virus_other\n",
      "Respiratory I-Virus_other\n",
      "Syndrome I-Virus_other\n",
      "coronavirus I-Virus_other\n",
      "( O\n",
      "SARS-CoV B-Virus_other\n",
      ") O\n",
      "outbreaks O\n",
      "in O\n",
      "2002-2003 O\n",
      "followed O\n",
      "by O\n",
      "Middle B-Virus_other\n",
      "East I-Virus_other\n",
      "Respiratory I-Virus_other\n",
      "Syndrome I-Virus_other\n",
      "CoV I-Virus_other\n",
      "( O\n",
      "MERS-CoV B-Virus_other\n",
      ") O\n",
      "\n",
      "\n",
      "This B-Virus_SARS-CoV-2\n",
      "decade's I-Virus_SARS-CoV-2\n",
      "first I-Virus_SARS-CoV-2\n",
      "CoV I-Virus_SARS-CoV-2\n",
      ", O\n",
      "named O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ", O\n",
      "emerged O\n",
      "from O\n",
      "Wuhan O\n",
      ", O\n",
      "China O\n",
      ", O\n",
      "and O\n",
      "declared O\n",
      "as O\n",
      "'Public O\n",
      "Health O\n",
      "Emergency O\n",
      "of O\n",
      "International O\n",
      "Concern' O\n",
      "on O\n",
      "January O\n",
      "30th O\n",
      ", O\n",
      "2020 O\n",
      "by O\n",
      "the O\n",
      "World O\n",
      "Health O\n",
      "Organization O\n",
      "( O\n",
      "WHO O\n",
      ") O\n",
      "\n",
      "\n",
      "As O\n",
      "on O\n",
      "February O\n",
      "4 O\n",
      ", O\n",
      "2020 O\n",
      ", O\n",
      "425 O\n",
      "deaths O\n",
      "reported O\n",
      "in O\n",
      "China O\n",
      "only O\n",
      "and O\n",
      "one O\n",
      "death O\n",
      "outside O\n",
      "China O\n",
      "( O\n",
      "Philippines O\n",
      ") O\n",
      "\n",
      "\n",
      "In O\n",
      "a O\n",
      "short O\n",
      "span O\n",
      "of O\n",
      "time O\n",
      ", O\n",
      "the O\n",
      "virus B-Virus_SARS-CoV-2\n",
      "spread O\n",
      "has O\n",
      "been O\n",
      "noted O\n",
      "in O\n",
      "24 O\n",
      "countries O\n",
      "\n",
      "\n",
      "The O\n",
      "zoonotic O\n",
      "transmission O\n",
      "( O\n",
      "animal-to-human O\n",
      ") O\n",
      "is O\n",
      "suspected O\n",
      "as O\n",
      "the O\n",
      "route O\n",
      "of O\n",
      "disease O\n",
      "origin O\n",
      "\n",
      "\n",
      "The O\n",
      "genetic O\n",
      "analyses O\n",
      "predict O\n",
      "bats O\n",
      "as O\n",
      "the O\n",
      "most O\n",
      "probable O\n",
      "source O\n",
      "of O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "though O\n",
      "further O\n",
      "investigations O\n",
      "needed O\n",
      "to O\n",
      "confirm O\n",
      "the O\n",
      "origin O\n",
      "of O\n",
      "the O\n",
      "novel B-Virus_SARS-CoV-2\n",
      "virus I-Virus_SARS-CoV-2\n",
      "\n",
      "\n",
      "The O\n",
      "ongoing O\n",
      "nCoV B-Virus_SARS-CoV-2\n",
      "outbreak O\n",
      "highlights O\n",
      "the O\n",
      "hidden O\n",
      "wild O\n",
      "animal O\n",
      "reservoir O\n",
      "of O\n",
      "the O\n",
      "deadly O\n",
      "viruses B-Virus_family\n",
      "and O\n",
      "possible O\n",
      "threat O\n",
      "of O\n",
      "spillover O\n",
      "zoonoses O\n",
      "as O\n",
      "well O\n",
      "\n",
      "\n",
      "The O\n",
      "successful O\n",
      "virus B-Virus_SARS-CoV-2\n",
      "isolation O\n",
      "attempts O\n",
      "have O\n",
      "made O\n",
      "doors O\n",
      "open O\n",
      "for O\n",
      "developing O\n",
      "better O\n",
      "diagnostics O\n",
      "and O\n",
      "effective O\n",
      "vaccines O\n",
      "helping O\n",
      "in O\n",
      "combating O\n",
      "the O\n",
      "spread O\n",
      "of O\n",
      "the O\n",
      "virus B-Virus_SARS-CoV-2\n",
      "to O\n",
      "newer O\n",
      "areas O\n",
      "\n",
      "\n",
      "Real-time O\n",
      "tentative O\n",
      "assessment O\n",
      "of O\n",
      "the O\n",
      "epidemiological O\n",
      "characteristics O\n",
      "of O\n",
      "novel B-Disease_COVID-19\n",
      "coronavirus I-Disease_COVID-19\n",
      "infections I-Disease_COVID-19\n",
      "in O\n",
      "Wuhan O\n",
      ", O\n",
      "China O\n",
      ", O\n",
      "as O\n",
      "at O\n",
      "22 O\n",
      "January O\n",
      "2020 O\n",
      "\n",
      "\n",
      "A O\n",
      "novel B-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "causing O\n",
      "severe B-Disease_COVID-19\n",
      "acute I-Disease_COVID-19\n",
      "respiratory I-Disease_COVID-19\n",
      "disease I-Disease_COVID-19\n",
      "emerged O\n",
      "recently O\n",
      "in O\n",
      "Wuhan O\n",
      ", O\n",
      "China O\n",
      "\n",
      "\n",
      "Information O\n",
      "on O\n",
      "reported O\n",
      "cases O\n",
      "strongly O\n",
      "indicates O\n",
      "human-to-human O\n",
      "spread O\n",
      ", O\n",
      "and O\n",
      "the O\n",
      "most O\n",
      "recent O\n",
      "information O\n",
      "is O\n",
      "increasingly O\n",
      "indicative O\n",
      "of O\n",
      "sustained O\n",
      "human-to-human O\n",
      "transmission O\n",
      "\n",
      "\n",
      "While O\n",
      "the O\n",
      "overall O\n",
      "severity O\n",
      "profile O\n",
      "among O\n",
      "cases O\n",
      "may O\n",
      "change O\n",
      "as O\n",
      "more O\n",
      "mild O\n",
      "cases O\n",
      "are O\n",
      "identified O\n",
      ", O\n",
      "we O\n",
      "estimate O\n",
      "a O\n",
      "risk O\n",
      "of O\n",
      "fatality O\n",
      "among O\n",
      "hospitalised O\n",
      "cases O\n",
      "at O\n",
      "14 O\n",
      "% O\n",
      "( O\n",
      "95 O\n",
      "% O\n",
      "confidence O\n",
      "interval O\n",
      ": O\n",
      "3 O\n",
      "\n",
      "\n",
      "9-32 O\n",
      "% O\n",
      ") O\n",
      "\n",
      "\n",
      "RNA O\n",
      "based O\n",
      "mNGS O\n",
      "approach O\n",
      "identifies O\n",
      "a O\n",
      "novel B-Virus_SARS-CoV-2\n",
      "human I-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "from O\n",
      "two O\n",
      "individual O\n",
      "pneumonia B-Disease_other\n",
      "cases O\n",
      "in O\n",
      "2019 O\n",
      "Wuhan O\n",
      "outbreak O\n",
      "\n",
      "\n",
      "From O\n",
      "December O\n",
      "2019 O\n",
      ", O\n",
      "an O\n",
      "outbreak O\n",
      "of O\n",
      "unusual O\n",
      "pneumonia B-Disease_other\n",
      "was O\n",
      "reported O\n",
      "in O\n",
      "Wuhan O\n",
      "with O\n",
      "many O\n",
      "cases O\n",
      "linked O\n",
      "to O\n",
      "Huanan O\n",
      "Seafood O\n",
      "Market O\n",
      "that O\n",
      "sells O\n",
      "seafood O\n",
      "as O\n",
      "well O\n",
      "as O\n",
      "live O\n",
      "exotic O\n",
      "animals O\n",
      "\n",
      "\n",
      "We O\n",
      "investigated O\n",
      "two O\n",
      "patients O\n",
      "who O\n",
      "developed O\n",
      "acute O\n",
      "respiratory B-Disease_other\n",
      "syndromes I-Disease_other\n",
      "after O\n",
      "independent O\n",
      "contact O\n",
      "history O\n",
      "with O\n",
      "this O\n",
      "market O\n",
      "\n",
      "\n",
      "The O\n",
      "two O\n",
      "patients O\n",
      "shared O\n",
      "common O\n",
      "clinical O\n",
      "features O\n",
      "including O\n",
      "fever B-Symptom\n",
      ", O\n",
      "cough B-Symptom\n",
      ", O\n",
      "and O\n",
      "multiple O\n",
      "ground-glass O\n",
      "opacities O\n",
      "in O\n",
      "the O\n",
      "bilateral O\n",
      "lung O\n",
      "field O\n",
      "with O\n",
      "patchy O\n",
      "infiltration O\n",
      "\n",
      "\n",
      "Here O\n",
      ", O\n",
      "we O\n",
      "highlight O\n",
      "the O\n",
      "use O\n",
      "of O\n",
      "a O\n",
      "low-input O\n",
      "metagenomic O\n",
      "next-generation O\n",
      "sequencing O\n",
      "( O\n",
      "mNGS O\n",
      ") O\n",
      "approach O\n",
      "on O\n",
      "RNA O\n",
      "extracted O\n",
      "from O\n",
      "bronchoalveolar O\n",
      "lavage O\n",
      "fluid O\n",
      "( O\n",
      "BALF O\n",
      ") O\n",
      "\n",
      "\n",
      "It O\n",
      "rapidly O\n",
      "identified O\n",
      "a O\n",
      "novel B-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "named O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "according O\n",
      "to O\n",
      "World O\n",
      "Health O\n",
      "Organization O\n",
      "announcement O\n",
      ") O\n",
      "which O\n",
      "was O\n",
      "the O\n",
      "sole O\n",
      "pathogens O\n",
      "in O\n",
      "the O\n",
      "sample O\n",
      "with O\n",
      "very O\n",
      "high O\n",
      "abundance O\n",
      "level O\n",
      "( O\n",
      "1 O\n",
      "\n",
      "\n",
      "5 O\n",
      "% O\n",
      "and O\n",
      "0 O\n",
      "\n",
      "\n",
      "62 O\n",
      "% O\n",
      "of O\n",
      "total O\n",
      "RNA O\n",
      "sequenced O\n",
      ") O\n",
      "\n",
      "\n",
      "The O\n",
      "entire O\n",
      "viral O\n",
      "genome O\n",
      "is O\n",
      "29 O\n",
      ", O\n",
      "881 O\n",
      "nt O\n",
      "in O\n",
      "length O\n",
      "( O\n",
      "GenBank O\n",
      "MN988668 O\n",
      "and O\n",
      "MN988669 O\n",
      ", O\n",
      "Sequence O\n",
      "Read O\n",
      "Archive O\n",
      "database O\n",
      "Bioproject O\n",
      "accession O\n",
      "PRJNA601736 O\n",
      ") O\n",
      "and O\n",
      "is O\n",
      "classified O\n",
      "into O\n",
      "β-coronavirus B-Virus_family\n",
      "genus O\n",
      "\n",
      "\n",
      "Phylogenetic O\n",
      "analysis O\n",
      "indicates O\n",
      "that O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "is O\n",
      "close O\n",
      "to O\n",
      "coronaviruses B-Virus_family\n",
      "( O\n",
      "CoVs B-Virus_family\n",
      ") O\n",
      "circulating O\n",
      "in O\n",
      "Rhinolophus O\n",
      "( O\n",
      "Horseshoe O\n",
      "bats O\n",
      ") O\n",
      ", O\n",
      "such O\n",
      "as O\n",
      "98 O\n",
      "\n",
      "\n",
      "7 O\n",
      "% O\n",
      "nucleotide O\n",
      "identity O\n",
      "to O\n",
      "partial O\n",
      "RdRp B-Protein\n",
      "gene O\n",
      "of O\n",
      "bat B-Virus_other\n",
      "coronavirus I-Virus_other\n",
      "strain I-Virus_other\n",
      "BtCoV/4991 I-Virus_other\n",
      "( O\n",
      "GenBank O\n",
      "KP876546 O\n",
      ", O\n",
      "370 O\n",
      "nt O\n",
      "sequence O\n",
      "of O\n",
      "RdRp B-Protein\n",
      "and O\n",
      "lack O\n",
      "of O\n",
      "other O\n",
      "genome O\n",
      "sequence O\n",
      ") O\n",
      "and O\n",
      "87 O\n",
      "\n",
      "\n",
      "9 O\n",
      "% O\n",
      "nucleotide O\n",
      "identity O\n",
      "to O\n",
      "bat B-Virus_other\n",
      "coronavirus I-Virus_other\n",
      "strain I-Virus_other\n",
      "bat-SL-CoVZC45 I-Virus_other\n",
      "and O\n",
      "bat-SL-CoVZXC21 B-Virus_other\n",
      "\n",
      "\n",
      "Evolutionary O\n",
      "analysis O\n",
      "based O\n",
      "on O\n",
      "ORF1a/1b O\n",
      ", O\n",
      "S O\n",
      ", O\n",
      "and O\n",
      "N O\n",
      "genes O\n",
      "also O\n",
      "suggests O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "is O\n",
      "more O\n",
      "likely O\n",
      "a O\n",
      "novel O\n",
      "CoV B-Virus_family\n",
      "independently O\n",
      "introduced O\n",
      "from O\n",
      "animals O\n",
      "to O\n",
      "humans O\n",
      "\n",
      "\n",
      "A O\n",
      "new B-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "associated O\n",
      "with O\n",
      "human O\n",
      "respiratory B-Disease_other\n",
      "disease I-Disease_other\n",
      "in O\n",
      "China O\n",
      "\n",
      "\n",
      "Emerging O\n",
      "infectious B-Disease_other\n",
      "diseases I-Disease_other\n",
      ", O\n",
      "such O\n",
      "as O\n",
      "severe B-Disease_other\n",
      "acute I-Disease_other\n",
      "respiratory I-Disease_other\n",
      "syndrome I-Disease_other\n",
      "( O\n",
      "SARS B-Disease_other\n",
      ") O\n",
      "and O\n",
      "Zika B-Disease_other\n",
      "virus I-Disease_other\n",
      "disease I-Disease_other\n",
      ", O\n",
      "present O\n",
      "a O\n",
      "major O\n",
      "threat O\n",
      "to O\n",
      "public O\n",
      "health1-3 O\n",
      "\n",
      "\n",
      "Despite O\n",
      "intense O\n",
      "research O\n",
      "efforts O\n",
      ", O\n",
      "how O\n",
      ", O\n",
      "when O\n",
      "and O\n",
      "where O\n",
      "new O\n",
      "diseases O\n",
      "appear O\n",
      "are O\n",
      "still O\n",
      "a O\n",
      "source O\n",
      "of O\n",
      "considerable O\n",
      "uncertainty O\n",
      "\n",
      "\n",
      "A O\n",
      "severe B-Disease_COVID-19\n",
      "respiratory I-Disease_COVID-19\n",
      "disease I-Disease_COVID-19\n",
      "was O\n",
      "recently O\n",
      "reported O\n",
      "in O\n",
      "Wuhan O\n",
      ", O\n",
      "Hubei O\n",
      "province O\n",
      ", O\n",
      "China O\n",
      "\n",
      "\n",
      "As O\n",
      "of O\n",
      "25 O\n",
      "January O\n",
      "2020 O\n",
      ", O\n",
      "at O\n",
      "least O\n",
      "1 O\n",
      ", O\n",
      "975 O\n",
      "cases O\n",
      "had O\n",
      "been O\n",
      "reported O\n",
      "since O\n",
      "the O\n",
      "first O\n",
      "patient O\n",
      "was O\n",
      "hospitalized O\n",
      "on O\n",
      "12 O\n",
      "December O\n",
      "2019 O\n",
      "\n",
      "\n",
      "Epidemiological O\n",
      "investigations O\n",
      "have O\n",
      "suggested O\n",
      "that O\n",
      "the O\n",
      "outbreak O\n",
      "was O\n",
      "associated O\n",
      "with O\n",
      "a O\n",
      "seafood O\n",
      "market O\n",
      "in O\n",
      "Wuhan O\n",
      "\n",
      "\n",
      "Here O\n",
      "we O\n",
      "study O\n",
      "a O\n",
      "single O\n",
      "patient O\n",
      "who O\n",
      "was O\n",
      "a O\n",
      "worker O\n",
      "at O\n",
      "the O\n",
      "market O\n",
      "and O\n",
      "who O\n",
      "was O\n",
      "admitted O\n",
      "to O\n",
      "the O\n",
      "Central O\n",
      "Hospital O\n",
      "of O\n",
      "Wuhan O\n",
      "on O\n",
      "26 O\n",
      "December O\n",
      "2019 O\n",
      "while O\n",
      "experiencing O\n",
      "a O\n",
      "severe O\n",
      "respiratory B-Disease_COVID-19\n",
      "syndrome I-Disease_COVID-19\n",
      "that O\n",
      "included O\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fever B-Symptom\n",
      ", O\n",
      "dizziness B-Symptom\n",
      "and O\n",
      "a O\n",
      "cough B-Symptom\n",
      "\n",
      "\n",
      "Metagenomic O\n",
      "RNA O\n",
      "sequencing4 O\n",
      "of O\n",
      "a O\n",
      "sample O\n",
      "of O\n",
      "bronchoalveolar O\n",
      "lavage O\n",
      "fluid O\n",
      "from O\n",
      "the O\n",
      "patient O\n",
      "identified O\n",
      "a O\n",
      "new B-Virus_SARS-CoV-2\n",
      "RNA I-Virus_SARS-CoV-2\n",
      "virus I-Virus_SARS-CoV-2\n",
      "strain I-Virus_SARS-CoV-2\n",
      "from I-Virus_SARS-CoV-2\n",
      "the I-Virus_SARS-CoV-2\n",
      "family I-Virus_SARS-CoV-2\n",
      "Coronaviridae I-Virus_SARS-CoV-2\n",
      ", O\n",
      "which O\n",
      "is O\n",
      "designated O\n",
      "here O\n",
      "'WH-Human B-Virus_SARS-CoV-2\n",
      "1' I-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "and O\n",
      "has O\n",
      "also O\n",
      "been O\n",
      "referred O\n",
      "to O\n",
      "as O\n",
      "'2019-nCoV' O\n",
      ") O\n",
      "\n",
      "\n",
      "Phylogenetic O\n",
      "analysis O\n",
      "of O\n",
      "the O\n",
      "complete O\n",
      "viral O\n",
      "genome O\n",
      "( O\n",
      "29 O\n",
      ", O\n",
      "903 O\n",
      "nucleotides O\n",
      ") O\n",
      "revealed O\n",
      "that O\n",
      "the O\n",
      "virus B-Virus_SARS-CoV-2\n",
      "was O\n",
      "most O\n",
      "closely O\n",
      "related O\n",
      "( O\n",
      "89 O\n",
      "\n",
      "\n",
      "1 O\n",
      "% O\n",
      "nucleotide O\n",
      "similarity O\n",
      ") O\n",
      "to O\n",
      "a O\n",
      "group O\n",
      "of O\n",
      "SARS-like B-Virus_family\n",
      "coronaviruses I-Virus_family\n",
      "( O\n",
      "genus O\n",
      "Betacoronavirus B-Virus_family\n",
      ", O\n",
      "subgenus O\n",
      "Sarbecovirus B-Virus_family\n",
      ") O\n",
      "that O\n",
      "had O\n",
      "previously O\n",
      "been O\n",
      "found O\n",
      "in O\n",
      "bats O\n",
      "in O\n",
      "China5 O\n",
      "\n",
      "\n",
      "This O\n",
      "outbreak O\n",
      "highlights O\n",
      "the O\n",
      "ongoing O\n",
      "ability O\n",
      "of O\n",
      "viral O\n",
      "spill-over O\n",
      "from O\n",
      "animals O\n",
      "to O\n",
      "cause O\n",
      "severe O\n",
      "disease O\n",
      "in O\n",
      "humans O\n",
      "\n",
      "\n",
      "Clinical O\n",
      "features O\n",
      "of O\n",
      "patients O\n",
      "infected O\n",
      "with O\n",
      "2019 B-Virus_SARS-CoV-2\n",
      "novel I-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "in O\n",
      "Wuhan O\n",
      ", O\n",
      "China O\n",
      "\n",
      "\n",
      "BACKGROUND O\n",
      ": O\n",
      "A O\n",
      "recent O\n",
      "cluster O\n",
      "of O\n",
      "pneumonia B-Symptom\n",
      "cases O\n",
      "in O\n",
      "Wuhan O\n",
      ", O\n",
      "China O\n",
      ", O\n",
      "was O\n",
      "caused O\n",
      "by O\n",
      "a O\n",
      "novel B-Virus_SARS-CoV-2\n",
      "betacoronavirus I-Virus_SARS-CoV-2\n",
      ", O\n",
      "the O\n",
      "2019 B-Virus_SARS-CoV-2\n",
      "novel I-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "\n",
      "\n",
      "We O\n",
      "report O\n",
      "the O\n",
      "epidemiological O\n",
      ", O\n",
      "clinical O\n",
      ", O\n",
      "laboratory O\n",
      ", O\n",
      "and O\n",
      "radiological O\n",
      "characteristics O\n",
      "and O\n",
      "treatment O\n",
      "and O\n",
      "clinical O\n",
      "outcomes O\n",
      "of O\n",
      "these O\n",
      "patients O\n",
      "\n",
      "\n",
      "METHODS O\n",
      ": O\n",
      "All O\n",
      "patients O\n",
      "with O\n",
      "suspected O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "were O\n",
      "admitted O\n",
      "to O\n",
      "a O\n",
      "designated O\n",
      "hospital O\n",
      "in O\n",
      "Wuhan O\n",
      "\n",
      "\n",
      "We O\n",
      "prospectively O\n",
      "collected O\n",
      "and O\n",
      "analysed O\n",
      "data O\n",
      "on O\n",
      "patients O\n",
      "with O\n",
      "laboratory-confirmed O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "infection O\n",
      "by O\n",
      "real-time O\n",
      "RT-PCR O\n",
      "and O\n",
      "next-generation O\n",
      "sequencing O\n",
      "\n",
      "\n",
      "Data O\n",
      "were O\n",
      "obtained O\n",
      "with O\n",
      "standardised O\n",
      "data O\n",
      "collection O\n",
      "forms O\n",
      "shared O\n",
      "by O\n",
      "WHO O\n",
      "and O\n",
      "the O\n",
      "International O\n",
      "Severe O\n",
      "Acute O\n",
      "Respiratory O\n",
      "and O\n",
      "Emerging O\n",
      "Infection O\n",
      "Consortium O\n",
      "from O\n",
      "electronic O\n",
      "medical O\n",
      "records O\n",
      "\n",
      "\n",
      "Researchers O\n",
      "also O\n",
      "directly O\n",
      "communicated O\n",
      "with O\n",
      "patients O\n",
      "or O\n",
      "their O\n",
      "families O\n",
      "to O\n",
      "ascertain O\n",
      "epidemiological O\n",
      "and O\n",
      "symptom O\n",
      "data O\n",
      "\n",
      "\n",
      "Outcomes O\n",
      "were O\n",
      "also O\n",
      "compared O\n",
      "between O\n",
      "patients O\n",
      "who O\n",
      "had O\n",
      "been O\n",
      "admitted O\n",
      "to O\n",
      "the O\n",
      "intensive O\n",
      "care O\n",
      "unit O\n",
      "( O\n",
      "ICU O\n",
      ") O\n",
      "and O\n",
      "those O\n",
      "who O\n",
      "had O\n",
      "not O\n",
      "\n",
      "\n",
      "FINDINGS O\n",
      ": O\n",
      "By O\n",
      "Jan O\n",
      "2 O\n",
      ", O\n",
      "2020 O\n",
      ", O\n",
      "41 O\n",
      "admitted O\n",
      "hospital O\n",
      "patients O\n",
      "had O\n",
      "been O\n",
      "identified O\n",
      "as O\n",
      "having O\n",
      "laboratory-confirmed O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "infection O\n",
      "\n",
      "\n",
      "Most O\n",
      "of O\n",
      "the O\n",
      "infected O\n",
      "patients O\n",
      "were O\n",
      "men O\n",
      "( O\n",
      "30 O\n",
      "[ O\n",
      "73 O\n",
      "% O\n",
      "] O\n",
      "of O\n",
      "41 O\n",
      ") O\n",
      "; O\n",
      "less O\n",
      "than O\n",
      "half O\n",
      "had O\n",
      "underlying O\n",
      "diseases O\n",
      "( O\n",
      "13 O\n",
      "[ O\n",
      "32 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      ", O\n",
      "including O\n",
      "diabetes B-Disease_other\n",
      "( O\n",
      "eight O\n",
      "[ O\n",
      "20 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      ", O\n",
      "hypertension B-Disease_other\n",
      "( O\n",
      "six O\n",
      "[ O\n",
      "15 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      ", O\n",
      "and O\n",
      "cardiovascular B-Disease_other\n",
      "disease I-Disease_other\n",
      "( O\n",
      "six O\n",
      "[ O\n",
      "15 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      "\n",
      "\n",
      "Median O\n",
      "age O\n",
      "was O\n",
      "49·0 O\n",
      "years O\n",
      "( O\n",
      "IQR O\n",
      "41·0-58·0 O\n",
      ") O\n",
      "\n",
      "\n",
      "27 O\n",
      "( O\n",
      "66 O\n",
      "% O\n",
      ") O\n",
      "of O\n",
      "41 O\n",
      "patients O\n",
      "had O\n",
      "been O\n",
      "exposed O\n",
      "to O\n",
      "Huanan O\n",
      "seafood O\n",
      "market O\n",
      "\n",
      "\n",
      "One O\n",
      "family O\n",
      "cluster O\n",
      "was O\n",
      "found O\n",
      "\n",
      "\n",
      "Common O\n",
      "symptoms O\n",
      "at O\n",
      "onset O\n",
      "of O\n",
      "illness O\n",
      "were O\n",
      "fever B-Symptom\n",
      "( O\n",
      "40 O\n",
      "[ O\n",
      "98 O\n",
      "% O\n",
      "] O\n",
      "of O\n",
      "41 O\n",
      "patients O\n",
      ") O\n",
      ", O\n",
      "cough B-Symptom\n",
      "( O\n",
      "31 O\n",
      "[ O\n",
      "76 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      ", O\n",
      "and O\n",
      "myalgia B-Symptom\n",
      "or O\n",
      "fatigue B-Symptom\n",
      "( O\n",
      "18 O\n",
      "[ O\n",
      "44 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      "; O\n",
      "less O\n",
      "common O\n",
      "symptoms O\n",
      "were O\n",
      "sputum B-Symptom\n",
      "production I-Symptom\n",
      "( O\n",
      "11 O\n",
      "[ O\n",
      "28 O\n",
      "% O\n",
      "] O\n",
      "of O\n",
      "39 O\n",
      ") O\n",
      ", O\n",
      "headache B-Symptom\n",
      "( O\n",
      "three O\n",
      "[ O\n",
      "8 O\n",
      "% O\n",
      "] O\n",
      "of O\n",
      "38 O\n",
      ") O\n",
      ", O\n",
      "haemoptysis B-Symptom\n",
      "( O\n",
      "two O\n",
      "[ O\n",
      "5 O\n",
      "% O\n",
      "] O\n",
      "of O\n",
      "39 O\n",
      ") O\n",
      ", O\n",
      "and O\n",
      "diarrhoea B-Symptom\n",
      "( O\n",
      "one O\n",
      "[ O\n",
      "3 O\n",
      "% O\n",
      "] O\n",
      "of O\n",
      "38 O\n",
      ") O\n",
      "\n",
      "\n",
      "Dyspnoea B-Symptom\n",
      "developed O\n",
      "in O\n",
      "22 O\n",
      "( O\n",
      "55 O\n",
      "% O\n",
      ") O\n",
      "of O\n",
      "40 O\n",
      "patients O\n",
      "( O\n",
      "median O\n",
      "time O\n",
      "from O\n",
      "illness O\n",
      "onset O\n",
      "to O\n",
      "dyspnoea B-Symptom\n",
      "8·0 O\n",
      "days O\n",
      "[ O\n",
      "IQR O\n",
      "5·0-13·0 O\n",
      "] O\n",
      ") O\n",
      "\n",
      "\n",
      "26 O\n",
      "( O\n",
      "63 O\n",
      "% O\n",
      ") O\n",
      "of O\n",
      "41 O\n",
      "patients O\n",
      "had O\n",
      "lymphopenia B-Symptom\n",
      "\n",
      "\n",
      "All O\n",
      "41 O\n",
      "patients O\n",
      "had O\n",
      "pneumonia B-Symptom\n",
      "with O\n",
      "abnormal O\n",
      "findings O\n",
      "on O\n",
      "chest O\n",
      "CT O\n",
      "\n",
      "\n",
      "Complications O\n",
      "included O\n",
      "acute B-Symptom\n",
      "respiratory I-Symptom\n",
      "distress I-Symptom\n",
      "syndrome I-Symptom\n",
      "( O\n",
      "12 O\n",
      "[ O\n",
      "29 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      ", O\n",
      "RNAaemia B-Symptom\n",
      "( O\n",
      "six O\n",
      "[ O\n",
      "15 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      ", O\n",
      "acute B-Symptom\n",
      "cardiac I-Symptom\n",
      "injury I-Symptom\n",
      "( O\n",
      "five O\n",
      "[ O\n",
      "12 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      "and O\n",
      "secondary B-Disease_other\n",
      "infection I-Disease_other\n",
      "( O\n",
      "four O\n",
      "[ O\n",
      "10 O\n",
      "% O\n",
      "] O\n",
      ") O\n",
      "\n",
      "\n",
      "13 O\n",
      "( O\n",
      "32 O\n",
      "% O\n",
      ") O\n",
      "patients O\n",
      "were O\n",
      "admitted O\n",
      "to O\n",
      "an O\n",
      "ICU O\n",
      "and O\n",
      "six O\n",
      "( O\n",
      "15 O\n",
      "% O\n",
      ") O\n",
      "died O\n",
      "\n",
      "\n",
      "Compared O\n",
      "with O\n",
      "non-ICU O\n",
      "patients O\n",
      ", O\n",
      "ICU O\n",
      "patients O\n",
      "had O\n",
      "higher O\n",
      "plasma O\n",
      "levels O\n",
      "of O\n",
      "IL2 B-Protein\n",
      ", O\n",
      "IL7 B-Protein\n",
      ", O\n",
      "IL10 B-Protein\n",
      ", O\n",
      "GSCF B-Protein\n",
      ", O\n",
      "IP10 B-Protein\n",
      ", O\n",
      "MCP1 B-Protein\n",
      ", O\n",
      "MIP1A B-Protein\n",
      ", O\n",
      "and O\n",
      "TNFα B-Protein\n",
      "\n",
      "\n",
      "INTERPRETATION O\n",
      ": O\n",
      "The O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "infection O\n",
      "caused O\n",
      "clusters O\n",
      "of O\n",
      "severe B-Disease_other\n",
      "respiratory I-Disease_other\n",
      "illness I-Disease_other\n",
      "similar O\n",
      "to O\n",
      "severe B-Virus_other\n",
      "acute I-Virus_other\n",
      "respiratory I-Virus_other\n",
      "syndrome I-Virus_other\n",
      "coronavirus I-Virus_other\n",
      "and O\n",
      "was O\n",
      "associated O\n",
      "with O\n",
      "ICU O\n",
      "admission O\n",
      "and O\n",
      "high O\n",
      "mortality O\n",
      "\n",
      "\n",
      "Major O\n",
      "gaps O\n",
      "in O\n",
      "our O\n",
      "knowledge O\n",
      "of O\n",
      "the O\n",
      "origin O\n",
      ", O\n",
      "epidemiology O\n",
      ", O\n",
      "duration O\n",
      "of O\n",
      "human O\n",
      "transmission O\n",
      ", O\n",
      "and O\n",
      "clinical O\n",
      "spectrum O\n",
      "of O\n",
      "disease O\n",
      "need O\n",
      "fulfilment O\n",
      "by O\n",
      "future O\n",
      "studies O\n",
      "\n",
      "\n",
      "FUNDING O\n",
      ": O\n",
      "Ministry O\n",
      "of O\n",
      "Science O\n",
      "and O\n",
      "Technology O\n",
      ", O\n",
      "Chinese O\n",
      "Academy O\n",
      "of O\n",
      "Medical O\n",
      "Sciences O\n",
      ", O\n",
      "National O\n",
      "Natural O\n",
      "Science O\n",
      "Foundation O\n",
      "of O\n",
      "China O\n",
      ", O\n",
      "and O\n",
      "Beijing O\n",
      "Municipal O\n",
      "Science O\n",
      "and O\n",
      "Technology O\n",
      "Commission O\n",
      "\n",
      "\n",
      "A O\n",
      "rapid O\n",
      "advice O\n",
      "guideline O\n",
      "for O\n",
      "the O\n",
      "diagnosis O\n",
      "and O\n",
      "treatment O\n",
      "of O\n",
      "2019 O\n",
      "novel O\n",
      "coronavirus O\n",
      "( O\n",
      "2019-nCoV O\n",
      ") O\n",
      "infected O\n",
      "pneumonia O\n",
      "( O\n",
      "standard O\n",
      "version O\n",
      ") O\n",
      "\n",
      "\n",
      "In O\n",
      "December O\n",
      "2019 O\n",
      ", O\n",
      "a O\n",
      "new B-Disease_COVID-19\n",
      "type I-Disease_COVID-19\n",
      "viral I-Disease_COVID-19\n",
      "pneumonia I-Disease_COVID-19\n",
      "cases O\n",
      "occurred O\n",
      "in O\n",
      "Wuhan O\n",
      ", O\n",
      "Hubei O\n",
      "Province O\n",
      "; O\n",
      "and O\n",
      "then O\n",
      "named O\n",
      "\"2019 O\n",
      "novel O\n",
      "coronavirus O\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "\" O\n",
      "by O\n",
      "the O\n",
      "World O\n",
      "Health O\n",
      "Organization O\n",
      "( O\n",
      "WHO O\n",
      ") O\n",
      "on O\n",
      "12 O\n",
      "January O\n",
      "2020 O\n",
      "\n",
      "\n",
      "For O\n",
      "it O\n",
      "is O\n",
      "a O\n",
      "never O\n",
      "been O\n",
      "experienced O\n",
      "respiratory B-Disease_other\n",
      "disease I-Disease_other\n",
      "before O\n",
      "and O\n",
      "with O\n",
      "infection O\n",
      "ability O\n",
      "widely O\n",
      "and O\n",
      "quickly O\n",
      ", O\n",
      "it O\n",
      "attracted O\n",
      "the O\n",
      "world's O\n",
      "attention O\n",
      "but O\n",
      "without O\n",
      "treatment O\n",
      "and O\n",
      "control O\n",
      "manual O\n",
      "\n",
      "\n",
      "For O\n",
      "the O\n",
      "request O\n",
      "from O\n",
      "frontline O\n",
      "clinicians O\n",
      "and O\n",
      "public O\n",
      "health O\n",
      "professionals O\n",
      "of O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "infected O\n",
      "pneumonia O\n",
      "management O\n",
      ", O\n",
      "an O\n",
      "evidence-based O\n",
      "guideline O\n",
      "urgently O\n",
      "needs O\n",
      "to O\n",
      "be O\n",
      "developed O\n",
      "\n",
      "\n",
      "Therefore O\n",
      ", O\n",
      "we O\n",
      "drafted O\n",
      "this O\n",
      "guideline O\n",
      "according O\n",
      "to O\n",
      "the O\n",
      "rapid O\n",
      "advice O\n",
      "guidelines O\n",
      "methodology O\n",
      "and O\n",
      "general O\n",
      "rules O\n",
      "of O\n",
      "WHO O\n",
      "guideline O\n",
      "development O\n",
      "; O\n",
      "we O\n",
      "also O\n",
      "added O\n",
      "the O\n",
      "first-hand O\n",
      "management O\n",
      "data O\n",
      "of O\n",
      "Zhongnan O\n",
      "Hospital O\n",
      "of O\n",
      "Wuhan O\n",
      "University O\n",
      "\n",
      "\n",
      "This O\n",
      "guideline O\n",
      "includes O\n",
      "the O\n",
      "guideline O\n",
      "methodology O\n",
      ", O\n",
      "epidemiological O\n",
      "characteristics O\n",
      ", O\n",
      "disease O\n",
      "screening O\n",
      "and O\n",
      "population O\n",
      "prevention O\n",
      ", O\n",
      "diagnosis O\n",
      ", O\n",
      "treatment O\n",
      "and O\n",
      "control O\n",
      "( O\n",
      "including O\n",
      "traditional O\n",
      "Chinese O\n",
      "Medicine O\n",
      ") O\n",
      ", O\n",
      "nosocomial B-Disease_other\n",
      "infection I-Disease_other\n",
      "prevention O\n",
      "and O\n",
      "control O\n",
      ", O\n",
      "and O\n",
      "disease O\n",
      "nursing O\n",
      "of O\n",
      "the O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "\n",
      "\n",
      "Moreover O\n",
      ", O\n",
      "we O\n",
      "also O\n",
      "provide O\n",
      "a O\n",
      "whole O\n",
      "process O\n",
      "of O\n",
      "a O\n",
      "successful O\n",
      "treatment O\n",
      "case O\n",
      "of O\n",
      "the O\n",
      "severe O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "infected O\n",
      "pneumonia O\n",
      "and O\n",
      "experience O\n",
      "and O\n",
      "lessons O\n",
      "of O\n",
      "hospital O\n",
      "rescue O\n",
      "for O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "infections O\n",
      "\n",
      "\n",
      "This O\n",
      "rapid O\n",
      "advice O\n",
      "guideline O\n",
      "is O\n",
      "suitable O\n",
      "for O\n",
      "the O\n",
      "first O\n",
      "frontline O\n",
      "doctors O\n",
      "and O\n",
      "nurses O\n",
      ", O\n",
      "managers O\n",
      "of O\n",
      "hospitals O\n",
      "and O\n",
      "healthcare O\n",
      "sections O\n",
      ", O\n",
      "community O\n",
      "residents O\n",
      ", O\n",
      "public O\n",
      "health O\n",
      "persons O\n",
      ", O\n",
      "relevant O\n",
      "researchers O\n",
      ", O\n",
      "and O\n",
      "all O\n",
      "person O\n",
      "who O\n",
      "are O\n",
      "interested O\n",
      "in O\n",
      "the O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "\n",
      "\n",
      "Drug O\n",
      "treatment O\n",
      "options O\n",
      "for O\n",
      "the O\n",
      "2019-new B-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "\n",
      "\n",
      "As O\n",
      "of O\n",
      "January O\n",
      "22 O\n",
      ", O\n",
      "2020 O\n",
      ", O\n",
      "a O\n",
      "total O\n",
      "of O\n",
      "571 O\n",
      "cases O\n",
      "of O\n",
      "the O\n",
      "2019-new B-Virus_SARS-CoV-2\n",
      "coronavirus I-Virus_SARS-CoV-2\n",
      "( O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      ") O\n",
      "have O\n",
      "been O\n",
      "reported O\n",
      "in O\n",
      "25 O\n",
      "provinces O\n",
      "( O\n",
      "districts O\n",
      "and O\n",
      "cities O\n",
      ") O\n",
      "in O\n",
      "China O\n",
      "\n",
      "\n",
      "At O\n",
      "present O\n",
      ", O\n",
      "there O\n",
      "is O\n",
      "no O\n",
      "vaccine O\n",
      "or O\n",
      "antiviral O\n",
      "treatment O\n",
      "for O\n",
      "human O\n",
      "and O\n",
      "animal O\n",
      "coronavirus B-Virus_family\n",
      ", O\n",
      "so O\n",
      "that O\n",
      "identifying O\n",
      "the O\n",
      "drug O\n",
      "treatment O\n",
      "options O\n",
      "as O\n",
      "soon O\n",
      "as O\n",
      "possible O\n",
      "is O\n",
      "critical O\n",
      "for O\n",
      "the O\n",
      "response O\n",
      "to O\n",
      "the O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "outbreak O\n",
      "\n",
      "\n",
      "Three O\n",
      "general O\n",
      "methods O\n",
      ", O\n",
      "which O\n",
      "include O\n",
      "existing O\n",
      "broad-spectrum O\n",
      "antiviral O\n",
      "drugs O\n",
      "using O\n",
      "standard O\n",
      "assays O\n",
      ", O\n",
      "screening O\n",
      "of O\n",
      "a O\n",
      "chemical O\n",
      "library O\n",
      "containing O\n",
      "many O\n",
      "existing O\n",
      "compounds O\n",
      "or O\n",
      "databases O\n",
      ", O\n",
      "and O\n",
      "the O\n",
      "redevelopment O\n",
      "of O\n",
      "new O\n",
      "specific O\n",
      "drugs O\n",
      "based O\n",
      "on O\n",
      "the O\n",
      "genome O\n",
      "and O\n",
      "biophysical O\n",
      "understanding O\n",
      "of O\n",
      "individual O\n",
      "coronaviruses B-Virus_family\n",
      ", O\n",
      "are O\n",
      "used O\n",
      "to O\n",
      "discover O\n",
      "the O\n",
      "potential O\n",
      "antiviral O\n",
      "treatment O\n",
      "of O\n",
      "human B-Virus_family\n",
      "pathogen I-Virus_family\n",
      "coronavirus I-Virus_family\n",
      "\n",
      "\n",
      "Lopinavir O\n",
      "/Ritonavir O\n",
      ", O\n",
      "Nucleoside O\n",
      "analogues O\n",
      ", O\n",
      "Neuraminidase O\n",
      "inhibitors O\n",
      ", O\n",
      "Remdesivir O\n",
      ", O\n",
      "peptide O\n",
      "( O\n",
      "EK1 O\n",
      ") O\n",
      ", O\n",
      "arbidol O\n",
      ", O\n",
      "RNA O\n",
      "synthesis O\n",
      "inhibitors O\n",
      "( O\n",
      "such O\n",
      "as O\n",
      "TDF O\n",
      ", O\n",
      "3TC O\n",
      ") O\n",
      ", O\n",
      "anti-inflammatory O\n",
      "drugs O\n",
      "( O\n",
      "such O\n",
      "as O\n",
      "hormones O\n",
      "and O\n",
      "other O\n",
      "molecules O\n",
      ") O\n",
      ", O\n",
      "Chinese O\n",
      "traditional O\n",
      "medicine O\n",
      ", O\n",
      "such O\n",
      "ShuFengJieDu O\n",
      "Capsules O\n",
      "and O\n",
      "Lianhuaqingwen O\n",
      "Capsule O\n",
      ", O\n",
      "could O\n",
      "be O\n",
      "the O\n",
      "drug O\n",
      "treatment O\n",
      "options O\n",
      "for O\n",
      "2019-nCoV B-Virus_SARS-CoV-2\n",
      "\n",
      "\n",
      "However O\n",
      ", O\n",
      "the O\n",
      "efficacy O\n",
      "and O\n",
      "safety O\n",
      "of O\n",
      "these O\n",
      "drugs O\n",
      "for O\n",
      "2019- B-Virus_SARS-CoV-2\n",
      "nCoV I-Virus_SARS-CoV-2\n",
      "still O\n",
      "need O\n",
      "to O\n",
      "be O\n",
      "further O\n",
      "confirmed O\n",
      "by O\n",
      "clinical O\n",
      "experiments O\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "IOB2_list =[]\n",
    "passages = soup.find_all('passage')\n",
    "### First we need to find all passages in the xml file which annotations are inside a \"passage\" block\n",
    "for i in passages:\n",
    "    ### Because I used split for spliting all the tokens I had to add space among words and puncutation signs \n",
    "    main_text = i.find('text').text\n",
    "    main_text = main_text.replace(',',' , ')\n",
    "    main_text = main_text.replace(':',' : ')\n",
    "    main_text = main_text.replace(';',' ; ')\n",
    "    main_text = main_text.replace('(',' ( ')\n",
    "    main_text = main_text.replace(')',' ) ')\n",
    "    main_text = main_text.replace('.',' . ')\n",
    "    main_text = main_text.replace('%',' % ')\n",
    "    main_text = main_text.replace('[',' [ ')\n",
    "    main_text = main_text.replace(']',' ] ')\n",
    "    \n",
    "    ##########################################################\n",
    "    ### Annotations blocks are NEs that the tag is presneted in the infon block \n",
    "    tokens = main_text.split()\n",
    "    annot = i.find_all('annotation')\n",
    "    words_per_pass = []\n",
    "    for j in annot:\n",
    "        info = j.find_all('infon')\n",
    "        for k in info:\n",
    "            if \"key=\\\"type\\\"\" in str(k):\n",
    "                words_per_pass.append([j.find('text').text, k.text])\n",
    "                \n",
    "    #######################################################\n",
    "    ## How \n",
    "    tokenCtr = 0\n",
    "    while tokenCtr < len(tokens):\n",
    "        isFound = False\n",
    "        for wpp in words_per_pass:\n",
    "            words = wpp[0].split()\n",
    "            tags  = wpp[1]\n",
    "            if tokens[tokenCtr:tokenCtr+len(words)] == words:\n",
    "                isFound = True\n",
    "                for tok in tokens[tokenCtr:tokenCtr+len(words)]:\n",
    "                    if tok is tokens[tokenCtr]:\n",
    "                        IOB2_list.append([tok,'B-'+ wpp[1]])\n",
    "                    else:\n",
    "                        IOB2_list.append([tok,'I-'+ wpp[1]])\n",
    "                \n",
    "                tokenCtr += len(words)-1\n",
    "                break\n",
    "        if not isFound:\n",
    "            IOB2_list.append([tokens[tokenCtr],'O'])\n",
    "        tokenCtr += 1\n",
    "    ########################################################            \n",
    "    ## This part is writing all the tokens by \"B\", \"I\", or \"O\" taggs to the output          \n",
    "with open(\"../files/Gold_std.IOB2\", 'w') as IOB2_file:           \n",
    "    for line in IOB2_list:\n",
    "        if line[0] == '.':\n",
    "            print('\\n',file=IOB2_file)\n",
    "            print('\\n')\n",
    "        else:\n",
    "            print(line[0], line[1], sep=' ',file=IOB2_file)\n",
    "            print(line[0], line[1], sep=' ')\n",
    "            \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
